### PR TITLE
[pallet-assets] add asset_exists(id: AssetId) function

### LINKED ISSUE #12444: [pallet-assets] add asset_exists(id: AssetId) function
### Is there an existing issue?

- [X] I have searched the existing issues

### Experiencing problems? Have you tried our Stack Exchange first?

- [X] This is not a support question.

### Description of bug

While working on an implementation of an Asset Registry for https://github.com/paritytech/trappist/issues/51, we needed to check whether an Asset exists.

We are currently doing the following, which feels a bit like a hack:
```
fn asset_exists(asset_id: AssetIdOf<T>) -> bool {
			!T::Assets::minimum_balance(asset_id).is_zero()
}
```

Ideally, `pallet-assets` should provide a function that does this, because it is basic functionality that is missing from the pallet.

### Steps to reproduce

_No response_

### PR DESCRIPTION
**Pull request by dharjeezy**
Checks if an asset exists via loose coupling(the `Inspect` trait).

closes #12444

### CODE REVIEW

File: frame/support/src/traits/tokens/fungibles.rs [hunk 1]
```diff
@@ -75,6 +75,9 @@ pub trait Inspect<AccountId> {
 		who: &AccountId,
 		amount: Self::Balance,
 	) -> WithdrawConsequence<Self::Balance>;
+
+	/// Determines if an `asset` exists
```
Comment (joepetrowski) [inline frame/support/src/traits/tokens/fungibles.rs:79]:
```suggestion
	/// Returns `true` if an `asset` exists.
```

**Review by ggwpez — APPROVED** [review]

**Comment by joepetrowski** [comment]
I think you should add this to `fungible::Inspect` too.

**Comment by dharjeezy** [comment]
> I think you should add this to `fungible::Inspect` too.

Alright

**Comment by dharjeezy** [comment]
> I think you should add this to `fungible::Inspect` too.

@joepetrowski in this case, are we going to introduce a new `type AssetId: AssetId;` since it isn't in the trait initally
?

**Comment by joepetrowski** [comment]
That is a great point. Seems obvious now, but looking at `fungible` I don't think it makes sense to add `AssetId` to it. In that case I'm OK with this PR.

**Review by joepetrowski — APPROVED** [review]

**Comment by ggwpez** [comment]
bot rebase

**Comment by ggwpez** [comment]
bot merge

**Comment by Polkadot-Forum** [comment]
This pull request has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/polkadot-release-analysis-v0-9-36/1529/1

**Pull request merged**